### DISCUSSION TITLE
Conflict Between Execution of main.pb.js and main.go

### DISCUSSION THREAD

**Discussion by Amirshp75** [pocketbase/pocketbase#5930] [Q&A] [answered]
Hi. I intend to write several API routes in `main.pb.js` due to my familiarity with JavaScript, and several routes in `main.go`. However, running these two files together creates a conflict.
The execution command for the first file is `pocketbase serve`, and for the second one, it's `go run main.go serve`.

I am currently developing an application that requires encrypting a text using a pre-existing key with the RSA algorithm. Additionally, I need to decrypt a text with this algorithm, which I could not implement in`main.pb.js`, but I managed to do it in Go. Now, I am forced to migrate the previous JavaScript routes to Go, which is difficult for me to translate.

1. Is there a way to use the crypto package in JavaScript`(main.pb.js)`?
2. Is the conflict in running these two files simultaneously inevitable?
3. How can both files be executed simultaneously, so that some routes are developed with JavaScript and others with Go?

Thank you.

**Comment by ganigeorgiev** [comment]
Try to be more specific. It is not clear what is the conflict you are talking about. If you are not able to explain it, then provide at least some code sample showing how do you initialize the JSVM in your Go code and what are the routes you are registering in both places.

Regarding loading external modules, please check https://pocketbase.io/docs/js-overview/#loading-modules.

**Comment by Amirshp75** [comment]
A: Let's assume the simple code below (from PocketBase documentation) is placed in the following file:
`~/pb_hooks/main.pb.js`

```
routerAdd("GET", "js/:name", (c) => {
    let name = c.pathParam("name")

    return c.json(200, { "message": "Hello js " + name })
})
```

With the route: `http://127.0.0.1:8090/js/hello`

B: Additionally, the following code is included in `main.go`:

```
package main

import (
	"log"
	"net/http"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
)

func main() {
	app := pocketbase.New()

	// serves static files from the provided public dir (if exists)
	app.OnBeforeServe().Add(func(e *core.ServeEvent) error {

		e.Router.GET("go/:name", func(c echo.Context) error {
			name := c.PathParam("name")

			return c.JSON(http.StatusOK, map[string]string{"message": "go " + name})
		} /* optional middlewares */)

		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}

}
```


With the route:`http://127.0.0.1:8090/go/amir`

The issue arises as follows:

Running the command `pocketbase serve` executes **only** the JS file, meaning it processes all routes from `pb_hooks` but cannot execute the routes from `main.go`.
Conversely, running the command `go run main.go serve` processes **only** the routes from`main.go` and cannot execute the JS routes from `pb_hooks`.
I would appreciate your advice on how to run both files simultaneously, enabling me to use routes from both JS and Go in the same project.

Thank you for your time and support.

**Reply by ganigeorgiev** [reply, marked as answer]
There is no conflict here. The JSVM must be explicitly enabled when creating your own executable. For example:

```go
jsvm.MustRegister(app, jsvm.Config{
    HooksWatch:    true,
    HooksPoolSize: 15,
})
```

As a reference, you can check [examples/base/main.go](https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go).
